app.rb

```diff
@@ -0,0 +1,101 @@
+require_relative 'book'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'classroom'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def display_books
+    @books.each do |book|
+      puts "Title: '#{book.title}', Author: #{book.author}"
+    end
+  end
+
+  def display_people
+    @people.each do |person|
+      puts "[#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [enter the number]: '
+    person_type = gets.chomp
+    if person_type != '1' && person_type != '2'
+      puts 'Person type not recognized. Back to the main menu.'
+      return
+    end
+
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp
+
+    if person_type == '1'
+      create_student(name, age)
+    elsif person_type == '2'
+      create_teacher(name, age)
+    end
+  end
+
+  def create_student(name, age)
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+    parent_permission = true if parent_permission == 'y'
+    parent_permission = false if parent_permission == 'n'
+    person = Student.new(age, nil, name: name, parent_permission: parent_permission)
+    @people << person
+    puts 'Student created successfully'
+  end
```

Comment:
- [ ] While trying to run your code and creating a student i came across the error shown in the image below when trying to pick an option for parent permission, becasue i am unable to create a person i am also not able to create a rental so i kindly suggest you fix this error in your code and resubmit :+1: :sparkles:
![pp](https://github.com/joramot/oop_school_library/assets/89055502/eea2042d-6e8c-4dd3-9807-e246b0fa87ae)